[Feature] Language and RTL settings menu page.

Mohtasim opened the pull request:
## Purpose
<!-- Describe the intention of the changes being proposed. What problem does it solve or functionality does it add? -->
* Changes the popup setting dialog box into a setting page
* Adds language selection functionality and RTL enable-disable option in the setting menu page 

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code
<!-- Add steps to run the tests suite and/or manually test -->
``` 
1. Launch the application from branch feature/localization-LanguageandRTLMenu
2. Experience the UI, persistence of the selected items and functionality 
3. To get the selected language setting value, call SettingsFeatures.isRTL(context) or SettingsFeatures.language(context)
```

## What to Check
Verify that the following are valid
* Tap setting from action bar to check the setting menu page
* Tap on Calling Settings label more than 5 times quickly to make appear the Language Setting menu

## Other Information
<!-- Add any other helpful information that may be needed here. -->
Enabling users to set language and RTL options from setting page

ahammer commented on azure-communication-ui/azure-communication-ui-demo-app/src/main/java/com/azure/android/communication/ui/callingcompositedemoapp/SettingsActivity.kt:
```diff
@@ -0,0 +1,64 @@
+package com.azure.android.communication.ui.callingcompositedemoapp
```
Add License

Mohtasim replied:
License added.

ahammer commented on azure-communication-ui/azure-communication-ui-demo-app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="utf-8"?>
```
License

ahammer replied:
on all new files

Mohtasim replied:
License added on the new files

ahammer commented on azure-communication-ui/azure-communication-ui-demo-app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,71 @@
+        android:layout_width="match_parent"
+        android:layout_height="2dp"
+        android:paddingTop="@dimen/azure_communication_ui_inner_space_padding"
+        android:background="@color/black" />
```
Hardcoded color, use attribute from theme, probably ?attr/colorOnSurface

Otherwise this will be non-contrasted in dark mode.

Mohtasim replied:
Updated the color according to theme.

ahammer reviewed (changes requested):
Add a companion object to SettingsActivity to get values/defaults from shared preferences.

e.g.

SettingsActivity.isRtl(context)
SettingsActivity.getSelectedLanguage(context)

Acceptance:
 - If in SharedPrefs, return those values
 - If not, return english defaults
 
 This way we'll be ready to pass the info into the Config/Composite after this PR is done.

Mohtasim commented:
All the requirement resolved and implemented. 
The commits are pushed as well. 
Implemented hidden language selection menu feature with secret taps.

ahammer commented on azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/configuration/SupportedLanguages.java:
```diff
@@ -0,0 +1,31 @@
+
+package com.azure.android.communication.ui.configuration;
+
+public enum SupportedLanguages {
```
Can you add, isRtl default value to enum, and update the settings UI to automatically select based on it?

iaulakh commented on azure-communication-ui/azure-communication-ui-demo-app/src/main/java/com/azure/android/communication/ui/callingcompositedemoapp/features/SettingsFeatures.kt:
```diff
@@ -0,0 +1,33 @@
+/**
```
We can follow same pattern as other files
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

Mohtasim replied:
Changed as suggested.

iaulakh commented on azure-communication-ui/azure-communication-ui-demo-app/src/main/java/com/azure/android/communication/ui/callingcompositedemoapp/features/SettingsFeatures.kt:
```diff
@@ -0,0 +1,33 @@
+    companion object {
+        var isLanguageFeatureEnabled: Boolean = false
+
+        // ---------------------- Language Features -------------------------------------
```
We can follow below pattern for comments (as other files) //

iaulakh commented on azure-communication-ui/azure-communication-ui-demo-app/src/main/java/com/azure/android/communication/ui/callingcompositedemoapp/SettingsActivity.kt:
```diff
@@ -0,0 +1,135 @@
+/**
```
We can follow same pattern as other files

Mohtasim replied:
Copyright message format changed.

iaulakh commented on azure-communication-ui/azure-communication-ui-demo-app/src/main/java/com/azure/android/communication/ui/callingcompositedemoapp/SettingsActivity.kt:
```diff
@@ -0,0 +1,135 @@
+    }
+}
+
+/**
```
We can use //

Mohtasim replied:
Changed according to the PR comment.

iaulakh commented on azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/configuration/SupportedLanguages.java:
```diff
@@ -0,0 +1,31 @@
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.configuration;
+
```
We can use ExpandableStringEnum? please search in code :)

iaulakh reviewed (changes requested):
Please run Android studio code formatting on all files, as well as ktlint.

iaulakh commented on azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/configuration/SupportedLanguages.java:
```diff
@@ -0,0 +1,89 @@
+     * @return returns the default or conventional isRTl boolean value
+     */
+    @Override
+    public Boolean getIsRTLDefaultValue() {
```
We may inject language as input? tomorrow for RTL language the check can return true/false as per language

iaulakh commented:
@Mohtasim we can change merge to "localization" branch?

ShaunaSong commented on CHANGELOG.md:
```diff
@@ -8,6 +8,7 @@
 - Localization support 
 - Update joining experience in setup view
+- Localization support 
 
```
Could you delete the second Localization support?
There is one before Update joining experience in setup view.
